src/System.Security.Cryptography.Hashing.Algorithms/tests/HmacTests.cs

```diff
@@ -31,7 +31,14 @@ protected HmacTests(byte[][] testKeys, byte[][] testData)
 
             using (HMAC hmac = Create())
             {
-                hmac.Key = _testKeys[testCaseId];
+                byte[] key = _testKeys[testCaseId];
+                hmac.Key = key;
+                Assert.NotSame(key, hmac.Key);
+                Assert.NotSame(hmac.Key, hmac.Key);
```

Comment:
This makes me wonder if we should also have a test that assigns Key to an array, changes the local array, and asserts that they are no longer Equal.  (The NotSame test here shows that each get_Key returns a copy, and my musing is showing that set_Key takes a copy, instead of maintaining the reference)